packages/sources/crypto-api-v2/src/endpoint/balance.ts

```diff
@@ -0,0 +1,33 @@
+import { balance } from '@chainlink/ea-factories'
+import { Requester } from '@chainlink/ea-bootstrap'
+import { Config } from '@chainlink/types'
+import { isCoinType, isChainType, TESTNET_BLOCKCHAINS, BLOCKCHAIN_NAME_MAP } from '.'
+
+export const Name = 'balance'
+
+const getBalanceURI = (address: string, chain: string, coin: string) => {
+  if (chain === 'testnet') chain = Requester.toVendorName(coin, TESTNET_BLOCKCHAINS) || chain
+  return `/v2/blockchain-data/${coin}/${chain}/addresses/${address}`
+}
+
+const getBalance: balance.GetBalance = async (account, config) => {
+  if (!account.coin) {
+    throw new Error(`Account ${account.address} is missing blockchain parameter`)
+  }
+  const coin = BLOCKCHAIN_NAME_MAP[account.coin.toLowerCase()]
+  const options = {
+    ...config.api,
+    url: getBalanceURI(account.address, account.chain as string, coin as string),
+  }
+  const response = await Requester.request(options)
+  const balance = response.data.data.item.confirmedBalance.amount
```

Comment:
We used to do some conversion in the CryptoApi V1 adapter.  We seemed to always assume that the balance is in BTC in the V1 API.  This isn't necessarily true in V2.